src/components/TodoHeader.tsx

```diff
@@ -0,0 +1,128 @@
+import React, {
+  useState, useRef, useEffect,
+} from 'react';
+import cn from 'classnames';
+import { Todo } from '../types/Todo';
+import { addTodos } from '../api/todos';
+import { ErrorMessage } from '../types/ErrorMessage';
+import { USER_ID } from '../utils/userId';
+
+type Props = {
+  todos: Todo[],
+  todo: Todo[],
+  setTodos: (todos: Todo[]) => void;
+  setTodosError: (error: ErrorMessage) => void;
+  tempTodos: Todo | null;
+  setTempTodos: (value: Todo | null) => void;
+  setIsLoading: (value: boolean | number) => void;
+  onTodoToggle: (value: Todo | null) => void;
+  setNewTodoLoad: (value: number) => void;
+  updatedTodos: boolean;
+};
+
+export const TodoHeader: React.FC<Props> = ({
+  todos,
+  todo,
+  setTodos,
+  setTodosError,
+  tempTodos,
+  setTempTodos,
+  setIsLoading,
+  onTodoToggle,
+  setNewTodoLoad,
+  updatedTodos,
+}) => {
+  const [todoTitle, setTodoTitle] = useState('');
+  const focusedInput = useRef<HTMLInputElement>(null);
+
+  const isActiveButton = todo.every(t => t.completed);
+
+  const handlerSubmitData = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!todoTitle.trim()) {
+      setTimeout(() => {
+        setTodosError(ErrorMessage.TitleShouldNotBeEmpty);
+      }, 3000);
+
+      return;
+    }
+
+    setIsLoading(true);
+    let id;
+
+    if (todos.length === 0) {
```

Comment:
```suggestion
    if (!todos.length) {
```

can be a bit simplify